cmd/compose/convert.go

```diff
@@ -145,7 +144,7 @@ func runConvert(ctx context.Context, streams api.Streams, backend api.Service, o
 		if err != nil {
 			return err
 		}
-		out = bufio.NewWriter(file)
+		out = io.Writer(file)
```

Comment:
Isn't `file` already a `io.Writer`?

Wondering if we're missing a `defer file.Close()`

Reply:
I guess since it's all writing at once, could also be `os.WriteFile(opts.Output, content, 0o666)` if a file is specified instead of stdout, which would take care of both creating, truncating, and writing to the file

Reply:
@thaJeztah The default output is `streams.Out()` and the changed line saved a `bufio.Writer` into `out` declared as `io.Writer`. `out` didn't have a method called `Close`, but I also tried to close the file. It didn't help. `bufio.Writer` needs to be flushed unless the output is larger the the buffer size. Closing the file wouldn't have solved this, since the content was not passed to the underlying writer. In this case, the `file`. Since the end of the `runConvert` method is using `fmt.Fprint` to write the content to a file or to stdout depending on the value of `out`,  I thought it was the easiest way to fix it, but you are right. I tried with this much shorter version to handle both stdout and the output file and worked:

```go
	if opts.Output != "" && len(content) > 0 {
		return os.WriteFile(opts.Output, content, 0o666)
	}
	_, err = fmt.Fprint(streams.Out(), string(content))
```

Should I force push the above code to the source branch?

Reply:
Thanks! Yes, that looks like the correct approach to me (but I'm not a maintainer on this repository, so it's possible that there's perhaps something I'm overlooking 🙈  😅)